### ISSUE TITLE
Deprecate `optuna.integration.KerasPruningCallback`.

### ISSUE THREAD

**Issue by crcrpar** [optuna/optuna#1643]
## TL; DR

Deprecate Keras Pruning Callback and examples of Keras:
- https://github.com/optuna/optuna/blob/master/optuna/integration/keras.py
- https://github.com/optuna/optuna/blob/master/examples/keras_simple.py
- https://github.com/optuna/optuna/blob/master/examples/pruning/keras_integration.py

## What to do

Basically, things need to be done are
1. decorate Keras pruning callback with [`deprecated` decorator](https://github.com/optuna/optuna/blob/master/optuna/_deprecated.py). `deprecated_version` would be 2.1.0 as of now. 
2. Referencing [the announcement from Keras](https://github.com/keras-team/keras/#multi-backend-keras-and-tfkeras) in the files mentioned above for users to know the information if they missed it. Specifically, mention the page in `text` of `deprecated` in the integration module and raise warn with the information in the examples.

related issue: https://github.com/optuna/optuna/issues/1479

Should you have any question, don't be shy to ping me 😃
